packages/extension-collaboration/src/collaboration.ts

```diff
@@ -173,6 +191,51 @@ export const Collaboration = Extension.create<CollaborationOptions>({
 
     const ySyncPluginInstance = ySyncPlugin(fragment, ySyncPluginOptions)
 
-    return [ySyncPluginInstance, yUndoPluginInstance]
+    return [
+      ySyncPluginInstance,
+      yUndoPluginInstance,
+      // Only add the filterInvalidContent plugin if content checking is enabled
+      this.editor.options.enableContentCheck
+      && new Plugin({
+        key: new PluginKey('filterInvalidContent'),
+        filterTransaction: tr => {
+          // Is this transaction from Yjs Sync Plugin?
+          if (isChangeOrigin(tr)) {
+
+            // When collaboration is disabled, prevent any sync transactions from being applied
+            if (this.storage.isCollaborationDisabled) {
+              // If collaboration is disabled, unregister the Yjs plugins to prevent further sync transactions
+              this.editor.unregisterPlugin(ySyncPluginKey)
+              this.editor.unregisterPlugin(yUndoPluginKey)
+              return false
+            }
+
+            // Did the doc actually change as a result of this transaction?
+            if (tr.docChanged) {
+
+              // Attempt to parse the content of the step
+              try {
+                const content = tr.doc.toJSON()
+
+                this.editor.schema.nodeFromJSON(content)
+              } catch (error) {
+                this.editor.emit('contentError', {
+                  error: error as Error,
+                  editor: this.editor,
+                  disableCollaboration: () => {
+                    this.storage.isCollaborationDisabled = true
+                    this.editor.unregisterPlugin(ySyncPluginKey)
+                    this.editor.unregisterPlugin(yUndoPluginKey)
```

Comment:
I think that the best way to disable collaboration here is actually just to disconnect the provider, we cannot actually disconnect the editor from reading changes in Y.js because it may not be only this plugin that is connected to it.

If we were to provide the provider as an option to the collaboration plugin we could be more certain that the changes CANNOT be synchronized to other users. Of course the caller could also easily implement this too, but it is both consistent with the current API on initialization and makes for an easy-to-use API.